src/helpers.coffee

```diff
@@ -244,6 +248,53 @@ exports.nameWhitespaceCharacter = (string) ->
     when '\t' then 'tab'
     else string
 
+exports.locationDataToBabylon = ({first_line, first_column, last_line, last_column, range}) ->
+  loc:
+    start:
+      line: first_line + 1
+      column: first_column
+    end:
+      line: last_line + 1
+      column: last_column + 1
+  range: [
+    range[0]
+    range[1] + 1
+  ]
+  start: range[0]
+  end: range[1] + 1
+
+exports.babylonLocationFields = ['loc', 'range', 'start', 'end']
+
+exports.isArray = isArray = (obj) -> Array.isArray obj
+exports.isFunction = (obj) -> Object::toString.call(obj) is '[object Function]'
+exports.isNumber = isNumber = (obj) -> Object::toString.call(obj) is '[object Number]'
+exports.isString = isString = (obj) -> Object::toString.call(obj) is '[object String]'
+exports.isBoolean = isBoolean = (obj) -> obj is yes or obj is no or Object::toString.call(obj) is '[object Boolean]'
+exports.isPlainObject = (obj) -> typeof obj is 'object' and !!obj and not isArray(obj) and not isNumber(obj) and not isString(obj) and not isBoolean(obj)
```

Comment:
I’m reluctant to add these very generic helpers to the codebase at this point. I think the general thinking so far has been to check for specific things that you’re expecting, rather than use broad brushes like `isBoolean`. In the places where these are being used, could we be more precise?